src/App.jsx

```diff
@@ -14,70 +15,68 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [goodState, setGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        <span>
+          {goodState === '' ? 'No goods selected' : `${goodState} is selected`}
+        </span>
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
+        {goodState === ''
+          ? ''
+          : (
             <button
-              data-cy="AddButton"
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              className="delete ml-3"
+              onClick={() => setGood('')}
+            />
+          )
+        }
+      </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+      <Table setGood={setGood} goodState={goodState} />
+    </main>
+  );
+};
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+export const Table = ({ setGood, goodState }) => (
+  <table className="table">
+    <tbody>
+      {goods.map(good => (
+        <Good setGood={setGood} good={good} goodState={goodState} key={good} />
+      ))}
+    </tbody>
+  </table>
+);
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+export const Good = ({ goodState, setGood, good }) => (
+  <tr
+    data-cy="Good"
+    className={`${goodState === good
+      ? 'has-background-success-light'
+      : ''}`}
+  >
+    <td>
+      <button
+        data-cy={`${goodState === good ? 'RemoveButton' : 'AddButton'}`}
+        type="button"
+        className={`button ${goodState === good
+          ? 'is-info'
+          : ''}`}
+        onClick={() => (
+          goodState === good ? setGood('') : setGood(good)
+        )}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)